### PR TITLE
doc: `google_re2` for type type.matcher.v3.RegexMatcher is not required any more

### PR DESCRIPTION
**Pull request by davinkevin**
Commit Message: Correction of a proto model to fix the actual state in envoy. The field was both required & deprecated. During execution, it was not required anymore. 
Additional Description: 
Risk Level: Low
Testing: None
Docs Changes: Correction of a proto model to fix the actual state in envoy. The field was both required & deprecated. During execution, it was not required anymore.
Release Notes: None
Platform Specific Features: None
Fixes https://github.com/envoyproxy/envoy/issues/26875

### CODE REVIEW

**Comment by adisuissa** [comment]
Please fix DCO and look at CI.

/wait

**Comment by phlax** [comment]
@davinkevin when you fix DCO if you also merge `main` it should fix the CI problems i think

**Comment by davinkevin** [comment]
Rebased and signed ✅

File: api/envoy/type/matcher/v3/regex.proto [hunk 1]
```diff
@@ -59,7 +59,7 @@ message RegexMatcher {
     // Google's RE2 regex engine.
     GoogleRE2 google_re2 = 1 [
       deprecated = true,
-      (validate.rules).message = {required: true},
+      (validate.rules).message = {required: false},
```
Comment (zhxie) [inline api/envoy/type/matcher/v3/regex.proto:62]: Why can't we just remove the validation rule?

**Review by zhxie — DISMISSED** [review]
LGTM, just with a small question.

File: api/envoy/type/matcher/v3/regex.proto (hunk 1 above)
Comment (phlax) [reply api/envoy/type/matcher/v3/regex.proto:62]: i thnk you can safely remove - i think this is just setting the default curently

Comment (davinkevin) [reply api/envoy/type/matcher/v3/regex.proto]: Done ✅

**Comment by phlax** [comment]
format diff is

```diff
diff --git a/api/envoy/type/matcher/v3/regex.proto b/api/envoy/type/matcher/v3/regex.proto
index c1fc749..10b3970 100644
--- a/api/envoy/type/matcher/v3/regex.proto
+++ b/api/envoy/type/matcher/v3/regex.proto
@@ -57,10 +57,8 @@ message RegexMatcher {
 
   oneof engine_type {
     // Google's RE2 regex engine.
-    GoogleRE2 google_re2 = 1 [
-      deprecated = true,
-      (envoy.annotations.deprecated_at_minor_version) = "3.0"
-    ];
+    GoogleRE2 google_re2 = 1
+        [deprecated = true, (envoy.annotations.deprecated_at_minor_version) = "3.0"];
   }
 
```

ci should post these - but looks like i may have broken that

**Comment by yanavlasov** [comment]
@davinkevin look like proto needs format fix, please https://dev.azure.com/cncf/envoy/_build/results?buildId=135304&view=logs&j=c95d62a5-6bb9-58f0-2a02-7575fd482a15&t=cbf2fcae-fb78-5fe6-095d-49113b477432&l=81

/wait

**Comment by davinkevin** [comment]
Updated, rebased and pushed again! I hope it's the one 😉

**Comment by davinkevin** [comment]
Fixed, I hope this one… is the good one! Sorry about this delay. 

I've rebased too in the process 😇

**Review by phlax — APPROVED** [review]
lgtm, thanks @davinkevin

**Pull request merged**